shared/teams/delete-team/index.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+import * as React from 'react'
+import * as Constants from '../../constants/teams'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+
+export type Props = {
+  onBack: () => void,
+  onDelete: () => void,
+  teamname: string,
+}
+
+const Header = (props: Props) => (
+  <>
+    <Kb.Avatar teamname={props.teamname} size={64} />
+    <Kb.Icon type="icon-team-delete-28" style={{marginRight: -60, marginTop: -20, zIndex: 1}} />
+    <Kb.Text style={styles.headerTeamname} type="BodySemibold">
+      {props.teamname}
+    </Kb.Text>
+  </>
+)
+
+type CheckboxesState = {|
+  box1: boolean,
+  box2: boolean,
+  box3: boolean,
+|}
+type CheckboxesProps = {|
+  onAllowDelete: boolean => void,
+|}
+
+class Checkboxes extends React.Component<CheckboxesProps, CheckboxesState> {
+  state = {
```

Comment:
i think its confusing to have this state and the state above in the parent where you're having to mange that by using didUpdate. I think it'd be simpler if this thing just took props for box1/2/3 and had click handlers from the parent and canDelete goes away and is just `this.state.box1 && this.state.box2 && this.state.box3`